rainier-core/src/main/scala/com/stripe/rainier/core/Discrete.scala

```diff
@@ -64,25 +69,40 @@ final case class Geometric(p: Real) extends Discrete {
 }
 
 /**
-  * Negative Binomial distribution with expectation `n(1-p)/p`
+  * Negative Binomial distribution with expectation `n*p/(1-p)`
   *
   * @param n Total number of failures
   * @param p Probability of success
   */
 final case class NegativeBinomial(n: Real, p: Real) extends Discrete {
-  val generator: Generator[Int] =
-    Generator.require(Set(n, p)) { (r, m) =>
+  val generator: Generator[Int] = {
+    val nbGenerator = Generator.require(Set(n, p)) { (r, m) =>
       (1 to m.toInt(n))
         .map({ x =>
-          Geometric(p).generator.get(r, m)
+          Geometric(1 - p).generator.get(r, m)
         })
         .sum
     }
+    val normalGenerator =
+      Normal(n * p / (1 - p), (n * p).pow(1.0 / 2.0) / (1 - p)).generator
+        .map(_.toInt.max(0))
+
+    Generator.from {
```

Comment:
would be slightly better as `normalGenerator.zip(nbGenerator).map{...}`

Reply:
```
    normalGenerator.zip(nbGenerator).map {
      case (normal, nbinom) =>
        val pDouble = ??
        val nDouble = ??

        if (pDouble < -100 / nDouble + 1 && pDouble > 100 / nDouble - .25) {
          normal
        } else { nbinom }
    }
```
Something like this? I'm not sure how to convert the `p` and `n` to double.

Reply:
ahh good point. Ok, let's leave it as is for now then.